File: 1155-number-of-dice-rolls-with-target-sum/1155-number-of-dice-rolls-with-target-sum.py
class Solution:
    def numRollsToTarget(self, n: int, k: int, target: int) -> int:
        dp = [[0 for _ in range(target+1)] for _ in range(n+1)]
        MOD = 1000000007
        dp[0][0] = 1

        for i in range(1, n+1):
            for j in range(1, target+1):
                for faces in range(1, k+1):
                    if j - faces >= 0:
                        dp[i][j] = (dp[i][j] + dp[i-1][j-faces]) % MOD

        return dp[n][target]

File: LeetCode/1155-number-of-dice-rolls-with-target-sum/1155-number-of-dice-rolls-with-target-sum.java
class Solution {
    public int numRollsToTarget(int n, int k, int target) {
        int MOD = 1000000007;
        int[][] dp = new int[n + 1][target + 1];
        dp[0][0] = 1;

        for (int i = 1; i <= n; i++) {
            for (int j = 1; j <= target; j++) {
                for (int faces = 1; faces <= k; faces++) {
                    if (j - faces >= 0) {
                        dp[i][j] = (dp[i][j] + dp[i - 1][j - faces]) % MOD;
                    }
                }
            }
        }
        return dp[n][target];
    }
}